examples/tutorials/send.js

```diff
@@ -1,19 +1,41 @@
 #!/usr/bin/env node
 
 var amqp = require('amqplib');
-var when = require('when');
 
-amqp.connect('amqp://localhost').then(function(conn) {
-  return when(conn.createChannel().then(function(ch) {
-    var q = 'hello';
-    var msg = 'Hello World!';
+function send(url, queueName, message) {
+  var connection;
 
-    var ok = ch.assertQueue(q, {durable: false});
-    
-    return ok.then(function(_qok) {
-      ch.sendToQueue(q, new Buffer(msg));
-      console.log(" [x] Sent '%s'", msg);
-      return ch.close();
-    });
-  })).ensure(function() { conn.close(); });;
-}).then(null, console.warn);
+  function createChannel(_connection) {
+    connection = _connection;
+    return connection.createChannel();
+  }
+
+  function setupQueue(channel) {
+    var queueConfig = {durable: false};
+    return channel.assertQueue(queueName, queueConfig);
+  }
+
+  function sendMsgToQueue(channel) {
+    channel.sendToQueue(queueName, new Buffer(message));
+  }
+  
+  function closeChannel(channel) {
+    return channel.close();
+  }
+
+  function closeConnection() {
+    if(connection) {
+      connection.close();
+    }
+  }
+  
+  return amqp.connect(url)
+    .then(createChannel)
+    .tap(setupQueue)
+    .tap(sendMsgToQueue)
```

Comment:
don't like seeing `tap` when native async/await becoming the new norm

Reply:
Keep in mind this PR was created 6 years ago. `async/await` wasn't the norm back then 😛